src/background/modules/AutocorrectHandler.js

```diff
@@ -26,6 +26,76 @@ let antipatterns = [];
 // Adapted from: https://github.com/mozilla/webextension-polyfill/blob/master/src/browser-polyfill.js
 const IS_CHROME = Object.getPrototypeOf(browser) !== Object.prototype;
 
+/**
+ * Traverse Trie tree of objects to create RegEx.
+ *
+ * @param {Object.<string, Object|boolean>} obj
+ * @returns {string}
+ */
+function pattern(obj) {
+    const alt = [];
+    const cc = [];
+
+    // Escape special characters
+    const regExSpecialChars = /[.*+?^${}()|[\]\\]/gu;
+
+    for (const char in obj) {
+        if (char) {
+            const achar = char.replace(regExSpecialChars, "\\$&");
+
+            if (!("" in obj[char] && Object.keys(obj[char]).length === 1)) {
+                const recurse = pattern(obj[char]);
+                alt.push(recurse + achar);
+            } else {
+                cc.push(achar);
+            }
+        }
+    }
+
+    if (cc.length) {
+        alt.push(cc.length === 1 ? cc[0] : `[${cc.join("")}]`);
+    }
+
+    let result = alt.length === 1 ? alt[0] : `(?:${alt.join("|")})`;
+
+    if ("" in obj) {
+        if (cc.length || alt.length > 1) {
+            result += "?";
+        } else {
+            result = `(?:${result})?`;
+        }
+    }
+
+    return result;
+}
+
+/**
+ * Convert autocorrections into Trie tree of objects.
+ *
+ * @param {string[]} arr
+ * @returns {string}
+ */
+function createRegEx(arr) {
+    const tree = {};
+
+    for (const s of arr.sort((a, b) => b.length - a.length)) {
```

Comment:
Also I am not sure, but should not that sorting be moved outside of the for loop, I vaguely remmeber JS would call it each time on iteration?

I may be wrong here though.

Reply:
OK, will do.